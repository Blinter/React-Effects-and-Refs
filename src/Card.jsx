// Card.jsx
import React from 'react';
import PropTypes from 'prop-types';
import './Deck.css';

function Card({ cardObject, rotation }) {
    const { value, suit, shown } = cardObject;

    return (
        <div
            className="card-container" data-testid="card-container"
        >
            <div className={`card-flip ${shown ? 'flipped' : ''}`} style={{ transform: rotation }}>
                <div className="card-face card-back">
                    {/* Back of the card */}
                </div>
                <div className="card-face card-front">
                    {shown && (
                        <img
                            className="card-image"
                            src={cardObject.png}
                            style={{
                                color: "transparent"
                            }}
                            aria-label={`${value} of ${suit}`}
                            alt={`${value} of ${suit}`}
                        />
                    )}
                </div>
            </div>
        </div>
    );
}

Card.propTypes = {
    cardObject: PropTypes.shape({
        png: PropTypes.string,
        image: PropTypes.string,
        value: PropTypes.string.isRequired,
        suit: PropTypes.string.isRequired,
        shown: PropTypes.bool.isRequired,
    }).isRequired,
    rotation: PropTypes.string.isRequired,
};

export default Card;