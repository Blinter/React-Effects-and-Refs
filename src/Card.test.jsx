// Deck.test.jsx

import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import axios from 'axios';
import Card from './Card';

jest.mock('axios');
describe('Card component', () => {
    const mockCardObject = {
        png: 'mock-image-url',
        image: 'mock-image-url',
        value: 'ACE',
        suit: 'SPADES',
        shown: true,
    };

    const mockRotation = 'rotateY(180deg)';

    beforeEach(() => {
        render(<Card
            cardObject={mockCardObject}
            rotation={mockRotation}
        />);
    });

    test('renders card container', () => {
        expect(screen.getByTestId('card-container')).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        const { container } = render(<Card
            cardObject={mockCardObject}
            rotation={mockRotation}
        />);
        expect(container).toMatchSnapshot();
    });

    test('applies rotation style', () => {
        const cardFlipElement = document.querySelector('.card-flip');
        expect(cardFlipElement).toHaveStyle(`transform: ${mockRotation}`);
    });
});