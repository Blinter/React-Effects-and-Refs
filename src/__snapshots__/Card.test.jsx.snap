// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches snapshot 1`] = `
<div>
  <div
    class="card-container"
    data-testid="card-container"
  >
    <div
      class="card-flip flipped"
      style="transform: rotateY(180deg);"
    >
      <div
        class="card-face card-back"
      />
      <div
        class="card-face card-front"
      >
        <img
          alt="ACE of SPADES"
          aria-label="ACE of SPADES"
          class="card-image"
          src="mock-image-url"
          style="color: transparent;"
        />
      </div>
    </div>
  </div>
</div>
`;
